autoload/fzf/vim.vim

```diff
@@ -396,9 +413,9 @@ function! fzf#vim#buffer_lines(...)
   let [query, args] = (a:0 && type(a:1) == type('')) ?
         \ [a:1, a:000[1:]] : ['', a:000]
   return s:fzf('blines', {
-  \ 'source':  s:buffer_lines(),
+  \ 'source':  empty(query) ? s:buffer_lines() : filter(s:buffer_lines(), 'v:val =~ "'.query.'"'),
```

Comment:
Filter function here should go inside `s:buffer_lines`, so that it is applied on unformatted, original lines.